src/client/Microsoft.Identity.Client/Instance/GenericAuthority.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Instance.Oidc;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class GenericAuthority : Authority
+    {
+
+        internal GenericAuthority(AuthorityInfo authorityInfo)
+            : base(authorityInfo)
+        {
+
+        }
+
+        internal override string TenantId => null;
+
+        internal override string GetTenantedAuthority(string tenantId, bool forceTenantless = false)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal override async Task<string> GetTokenEndpointAsync(RequestContext requestContext)
+        {
+            var configuration = await OidcRetrieverWithCache.GetOidcAsync(
+                AuthorityInfo.CanonicalAuthority.AbsoluteUri, requestContext)
+                    .ConfigureAwait(false);
+
+            return configuration.TokenEndpoint;
+        }
+
+        internal override async Task<string> GetAuthorizationEndpointAsync(
+            RequestContext requestContext)
+        {
+            OidcMetadata configuration = await OidcRetrieverWithCache.GetOidcAsync(
+               AuthorityInfo.CanonicalAuthority.AbsoluteUri, requestContext)
+                .ConfigureAwait(false);
+
+            return configuration.AuthorizationEndpoint;
+        }
+
+        internal override Task<string> GetDeviceCodeEndpointAsync(RequestContext requestContext)
+        {
+            // prevents device_code flow which requires knowledge of the device_authorization_endpoint.
+            throw new NotImplementedException();
```

Comment:
If this is going to be enabled for public clients then I think it makes sense to throw a more actionable error here so developers have a clear understanding of when this should not be used with generic authorities. This can also be handled somewhere higher up the chain of calls when the application is being configured. More of a suggestion for now.

Reply:
I'll move it to CCA. The API is experimental, so not too worried about not being ready everywhere.